app/controllers/transactions/transaction-list.controller.js

```diff
@@ -25,18 +25,28 @@ module.exports = async function showTransactionList (req, res, next) {
     return next(new Error('Invalid search'))
   }
 
+  let noTransactionSearchResults = {
+    total: 0,
+    count: 0,
+    page: 1,
+    results: [],
+    _links: {},
+    filters: {},
+  }
```

Comment:
The `let noTransactionSearchResults` in the controller is the default initialisation of what the Ledger API endpoint returns i.e. `uk.gov.pay.ledger.transaction.model.TransactionSearchResponse`, when a transaction search is done. Setting up the stub simplifies the logic in transaction-view.js buildPaymentList(....). Otherwise, we would be writing a new and big function to populate/persist all the filter values when the search fails date validation.

Reply:
As discussed, the existing logic is cool, I have renamed the variable as declared as a constant.